From 1ada31065a54d78d07bb8102cd51b1374526844c Mon Sep 17 00:00:00 2001
From: Frank Earl <frank@earlconsult.com>
Date: Tue, 19 Mar 2019 08:06:55 -0500
Subject: Don't fail to have a value in the structure for the phy...

The process may leave nulls in the structure for a given wiphy.
Not all devices may reside in the HWDB on a target but have
valid driver modules and firmware.  Push an UNKNOWN when you
can't find vendor or model info rather than lazily keeping
it EMPTY, which will cause follow-on controllers like connman
to CHOKE.

diff --git a/src/wiphy.c b/src/wiphy.c
index 6c1cf38..19aa498 100644
--- a/src/wiphy.c
+++ b/src/wiphy.c
@@ -711,6 +711,13 @@ static void wiphy_register(struct wiphy *wiphy)
 		l_hwdb_lookup_free(entries);
 	}
 
+    /* Ensure you have the above parameters filled with **SOMETHING** 
+       instead of passing a NULL back to other callers... */
+    if (wiphy->model_str == NULL)
+        wiphy->model_str = l_strdup("UNKNOWN");
+    if (wiphy->vendor_str == NULL)
+        wiphy->vendor_str = l_strdup("UNKNOWN");
+
 	if (!l_dbus_object_add_interface(dbus, wiphy_get_path(wiphy),
 					IWD_WIPHY_INTERFACE, wiphy))
 		l_info("Unable to add the %s interface to %s",
